data/resources/ui/content-message-document.ui

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <template class="ContentMessageDocument" parent="ContentMessageBase">
+    <property name="layout-manager">
+      <object class="GtkBoxLayout">
+        <property name="orientation">vertical</property>
+      </object>
+    </property>
+    <style>
+      <class name="message-bubble"/>
+    </style>
+    <child>
+      <object class="GtkLabel" id="sender_label">
+        <property name="ellipsize">end</property>
+        <property name="single-line-mode">True</property>
+        <property name="xalign">0</property>
+        <style>
+          <class name="sender-text"/>
+        </style>
+      </object>
+    </child>
+    <child>
+      <object class="GtkBox">
+        <property name="spacing">6</property>
+        <child>
+          <object class="GtkButton" id="file_status_button">
+            <style>
+              <class name="circular"/>
+            </style>
+            <property name="valign">start</property>
+          </object>
+        </child>
```

Comment:
I just discovered that we might run into two small usability problems:
1. The button rather looks like a normal icon instead of a button. So it is not immediately clear that it is clickable.
2. The hit target (button) is rather small. In the official client, the user can click the whole message to download or to view

I suggest the following:
1. Exchange the button with a normal icon
2. For the whole message except the `content_label`
    2.1.  Set `obj.set_cursor_from_name(Some("hand2"));`
    2.2. Use a `GtkGestureClick` like in https://github.com/melix99/telegrand/blob/088f46766c24592630308ba67d92a1659a21eb77/src/session/content/message_row/base.rs#L15

Reply:
I think, you can replace your code, where you set the button's signal handler easily. You can setup everything in rust code:
```rust
        let click = gtk::GestureClick::new();
        click.connect_pressed(|_, _, _, _| {

        });
        self.add_controller(&click);
        self.remove_controller(&click);
```

Reply:
Ok, I will try this tomorrow

Reply:
I think that button needs to be button:
- It can be moved to separate widget and used in images and videos
- In telegram when download started message area isn't clickable and you need click that button to cancel loading 

I've solved first problem implemented hover style

[Kooha-2022-09-11-12-27-53.webm](https://user-images.githubusercontent.com/63008755/189520772-df2df707-90aa-4e7f-bd00-47370e38a40d.webm)


In telegram when you hover message cursor looks like hand
![image](https://user-images.githubusercontent.com/63008755/189520920-1168e6fa-1a64-4916-ae4e-dc73ab20e46e.png)
gnome use this cursor only for links

So I think that I need to add click gesture to box with button and labels that is clickable when file CanBeDownloaded or Downloaded and it will 'click' button and add hover style to box when it's clickable

Reply:
[Kooha-2022-09-11-14-06-20.webm](https://user-images.githubusercontent.com/63008755/189524163-7bddcd05-0708-4997-8f86-ea02517e93a1.webm)
Now aall works